### DISCUSSION TITLE
Image resizing while respecting aspect ratio

### DISCUSSION THREAD

**Discussion by bennet-org** [tinacms/tinacms#3281] [Ideas]
Hi all,

I'm using the [tina-cloud-starter](https://github.com/tinacms/tina-cloud-starter) template and have a question about embedding images in blog posts. Main criteria for me: content editors need to be able to upload original high-res photos to the media manager, and the production site needs to serve suitably resized versions while respecting original aspect ratio.

Using the repo's standard `img` component with [repo-based media](https://tina.io/docs/reference/media/repo-based/) works, but large image files aren't downsized. Attempting to use Netlify's [image optimisation & Git LFS](https://docs.netlify.com/large-media/transform-images/) doesn't seem to play nicely with Tina Cloud. I'd rather use `next/image`, as [suggested](https://tina.io/docs/reference/media/repo-based/#nextimage) in the Tina docs. However this requires either passing in width and height (which aren't available to the component, as far as I can tell) or using `layout=fill` and a fixed aspect ratio. The fixed aspect ratio option works fine, but obviously means images are sometimes badly cropped.

We need a way to respect aspect ratio _and_ resize appropriately, perhaps by passing image metadata to the component so aspect ratio can be calculated. @konsumer suggested a workaround [here](https://github.com/tinacms/tinacms/discussions/3179), but this feels like something which would benefit content teams as an out-the-box feature.

Thanks!

Tom

**Comment by jamespohalloran** [comment]
Hey Tom, if Tina was to return some image metadata along with the image-url from our API (including width and height), would that solve this issue for you, for the `next/image` component?

**Reply by bennet-org** [reply]
Hey @jamespohalloran. Yep, I think that would solve the problem! We could then pass in width and height params to the `next/image` component, avoiding the need to crop using `layout=fill`.
